mmseg/apis/ote/apis/segmentation/inference_task.py

```diff
@@ -265,38 +261,54 @@ def _infer_segmentor(model: torch.nn.Module, config: Config, dataset: DatasetEnt
         else:
             eval_model = MMDataCPU(model)
 
-        eval_predictions = []
-        feature_vectors = []
+        feature_vector = None
 
         def dump_features_hook(mod, inp, out):
             with torch.no_grad():
                 feature_map = out[0]
-                feature_vector = torch.nn.functional.adaptive_avg_pool2d(feature_map, (1, 1))
-                assert feature_vector.size(0) == 1
-
-            feature_vectors.append(feature_vector.view(-1).detach().cpu().numpy())
+                feature_map = torch.nn.functional.adaptive_avg_pool2d(feature_map, (1, 1))
+                assert feature_map.size(0) == 1
+            nonlocal feature_vector
+            feature_vector = feature_map.view(-1).detach().cpu().numpy()
 
         def dummy_dump_features_hook(mod, inp, out):
-            feature_vectors.append(None)
+            pass
 
         hook = dump_features_hook if dump_features else dummy_dump_features_hook
 
+        stats = defaultdict(float)
+            
         # Use a single gpu for testing. Set in both mm_val_dataloader and eval_model
         with eval_model.module.backbone.register_forward_hook(hook):
-            for data in mm_val_dataloader:
+            for i, (data, dataset_item) in enumerate(zip(mm_val_dataloader, dataset)):
                 with torch.no_grad():
                     result = eval_model(return_loss=False, output_logits=output_logits, **data)
-                eval_predictions.extend(result)
+                assert len(result) == 1
+                if eval:
+                    for k, v in mm_val_dataset.compute_stat_per_frame(result[0], frame_id=i).items():
+                        stats[k] += v
```

Comment:
I just tried to make changes without changing behavior.

It is set here https://github.com/openvinotoolkit/mmsegmentation/blob/56366df858d6f84641c0b3595ec03efe3bfda446/mmseg/apis/ote/apis/segmentation/train_task.py#L110 as model performance. Moreover mmseg's mDice differs from Dice score from SDK

Looks like a bug, right @druzhkov-paul ?

@evgeny-izutov ?

Reply:
@Ilya-Krylov I don't second your opinion on this.

I guess, we can set `score` to zero or anything dummy and leave internal metric only for best checkpoint selection.

Reply:
@evgeny-izutov can me remove these computations? what do you think?

Reply:
@Ilya-Krylov -- I think it's mostly the question on how the [line](https://github.com/openvinotoolkit/mmsegmentation/blob/56366df858d6f84641c0b3595ec03efe3bfda446/mmseg/apis/ote/apis/segmentation/train_task.py#L115) is used later. For me it's good decision because it removes the ambiguity of metric computations inside/outside mmseg.

Reply:
removed, set score to 0 and metric name to "None"